src/client/opamClient.ml

```diff
@@ -1116,7 +1116,7 @@ let install_t t ?ask ?(ignore_conflicts=false) ?(depext_only=false)
   let requested =
     OpamPackage.Name.Set.of_list (List.rev_map fst (atoms @ deps_atoms))
   in
-  let packages = OpamPackage.packages_of_names t.packages requested in
+  let packages = OpamFormula.packages_of_atoms t.packages (atoms @ deps_atoms) in
```

Comment:
Could you show an example of the bug? I'm a bit surprise this change is not equivalent

Reply:
The example is the same as in #4796. The difference between `packages_of_names` and `packages_of_atoms` is that the former does not take into account the version constraints of the atoms. Hence, it will allow any version of the requested package to be installed instead of a specifically requested version.

Reply:
ah i see.